synapse/storage/util/id_generators.py

```diff
@@ -727,7 +764,10 @@ def _mark_id_as_finished(self, next_id: int) -> None:
                     curr, new_cur, self._max_position_of_local_instance
                 )
 
-            self._add_persisted_position(next_id)
+            # TODO Can we call this for just the last position or somehow batch
+            # _add_persisted_position.
```

Comment:
I think this is a question for @erikjohnston -- `_add_persisted_position` seems fairly heavy and if we could optimize what we call it with (or take multiple IDs at once) I think that'd be a win.

Reply:
I think we certainly could make make it take multiple IDs? It does expect to be called with every position (though I don't think anything would break, just be less efficient)

Reply:
I guess if it has to be called w/ every position maybe it isn't a huge deal...